app/src/main/java/comp3350/mealbuddy/persistence/DataAccessStub.java

```diff
@@ -0,0 +1,153 @@
+    }
+    public String name;
+
+    //the "databases"
+    private ArrayList<Edible> edibles;
+    private ArrayList<Account> accounts;
+    private ArrayList<Exercise> exercises;
+
+    public DataAccessStub(String name){
+        this.name = name;
+    }
+
+    public void open() {
+        //initialize the arrays
+        edibles = new ArrayList<>();
+        accounts = new ArrayList<>();
+        exercises = new ArrayList<>();
+
+        //initialize the data
+        initAccounts();
+        initEdibles();
+        initExercises();
+        System.out.println("Opened Database " + name);
+    }
+
+    public void close(){
+        System.out.println("Closed database " + name);
+    }
+
+    private void initEdibles(){
+        Food edible;
+        Meal meal = new Meal("Burger", new ArrayList<String>());
+        Object[][] foods = {
+            {"Beef Burger", new ArrayList<String>()},
+            {"Cheese", new ArrayList<>(Arrays.asList("dairy", "vegetarian"))},
+            {"Burger Bun", new ArrayList<>(Arrays.asList("vegetarian", "vegan"))},
+        };
+        for (Object[] food : foods){
+            edible = new Food(
+                    (String)food[0],
+                    (ArrayList)food[1]
+            );
+            edibles.add(edible);
+            meal.updateFood(edible, 1);
+        }
+    }
+
+    private void initAccounts(){
+        Account acc;
+        UserInfo userInfo;
+        //format of a user:
+            //Full Name | username | password | weight (in lbs) | height (in cm) | Activity Level | Sex
+        Object[][] users = {
+            {"Ned Stark", "ned_stark123", "starksrule", 200.0, 195.0, UserInfo.ActivityLevel.MEDIUM, UserInfo.Sex.MALE},
+            {"Jamie Lannister", "k1ngsl4yer", "alwayspaymydebts", 230.0, 200.0, UserInfo.ActivityLevel.HIGH, UserInfo.Sex.MALE},
+            {"Daenerys Targaryen", "motherOfDragzz", "123123", 150.0, 160.0, UserInfo.ActivityLevel.MEDIUM, UserInfo.Sex.FEMALE},
+            {"Catelyn Stark", "stoneheart", "909090", 154.4, 160.0, UserInfo.ActivityLevel.LOW, UserInfo.Sex.FEMALE},
+            {"Admin", "admin", "group5", 14.2, 400.3, UserInfo.ActivityLevel.LOW, UserInfo.Sex.MALE}
+        };
+
+        for (Object[] user : users){
+            userInfo = new UserInfo(
+                    (String)user[0],
+                    (String)user[1],
+                    (String)user[2],
+                    (double)user[3],
+                    (double)user[4],
+                    (UserInfo.ActivityLevel)user[5],
+                    (UserInfo.Sex)user[6]
+            );
+            acc = new Account(userInfo);
+            accounts.add(acc);
+        }
+    }
+
+    private void initExercises(){
+        String[] exerciseList = {
+            "Outdoor Run", "Bench Press", "Push Ups", "Sit Ups"
+        };
+        for (String e : exerciseList){
+            exercises.add(new Exercise(e));
+        }
+    }
+
+    public void addToDB(Database_t DT, Object o){
```

Comment:
This is cool how the stub is done

Reply:
I could've reduced it further with making a variable for like "activeList" or something, but at that point it would get dangerous. Figured this would be a good way to keep it looking good but also simple

Reply:
Yeah really good.